examples/contrastive-image-text/clip_media_pipe.py

```diff
@@ -0,0 +1,162 @@
+
+
+class read_image_text_from_dataset(MediaReaderNode):
+    """
+    Class defining read image/text from directory node.
+
+    """
+
+    def __init__(self, name, guid, device, inputs, params, cparams, node_attr):
+        super().__init__(name, guid, device, inputs, params, cparams, node_attr)
+        self.meta_dtype = params["label_dtype"]
+        self.dataset = params["dataset"]
+        self.epoch = 0
+
+        self.num_imgs_slice = len(ClipMediaPipe.batch_sampler.sampler)
+        self.num_batches_slice = len(ClipMediaPipe.batch_sampler)
+        print("Finding largest file ...")
+        self.max_file = get_max_file(self.dataset["image_path"])
+        print("largest file is ", self.max_file)
+
+    def set_params(self, params):
+        self.batch_size = params.batch_size
+
+    def gen_output_info(self):
+        out_info = []
+        o = opnode_tensor_info(dtype.NDT, np.array([self.batch_size], dtype=np.uint32), "")
+        out_info.append(o)
+        o = opnode_tensor_info(
+            self.meta_dtype, np.array([self.dataset.text_max_length, self.batch_size], dtype=np.uint32), ""
+        )
+        out_info.append(o)
+        o = opnode_tensor_info(
+            self.meta_dtype, np.array([self.dataset.text_max_length, self.batch_size], dtype=np.uint32), ""
+        )
+        out_info.append(o)
+        return out_info
+
+    def get_largest_file(self):
+        return self.max_file
+
+    def get_media_output_type(self):
+        return readerOutType.FILE_LIST
+
+    def __len__(self):
+        return self.num_batches_slice
+
+    def __iter__(self):
+        self.iter_loc = 0
+        self.batch_sampler_iter = iter(ClipMediaPipe.batch_sampler)
+        self.epoch += 1
+        return self
+
+    def __next__(self):
+        if self.iter_loc > (self.num_imgs_slice - 1):
+            raise StopIteration
+
+        data_idx = next(self.batch_sampler_iter)
+        data = self.dataset.__getitems__(data_idx)
+        img_list = []
+
+        input_id_list = np.zeros(shape=(self.batch_size, self.dataset.text_max_length), dtype=self.meta_dtype)
+        attention_mask_list = np.zeros(shape=(self.batch_size, self.dataset.text_max_length), dtype=self.meta_dtype)
+        for i, x in enumerate(data):
+            img_list.append(x["image_path"])
+            input_id_list[i, :] = x["input_ids"]
+            attention_mask_list[i, :] = x["attention_mask"]
+
+        self.iter_loc = self.iter_loc + self.batch_size
+
+        return img_list, input_id_list, attention_mask_list
+
+
+read_image_text_from_dataset_params = {
+    "label_dtype": dtype.UINT64,
+    "dataset": None,
+}
+schema.add_operator(
+    "ClipDataReader",
+    None,
+    0,
+    0,
+    [],
+    3,
+    read_image_text_from_dataset_params,
+    None,
+    read_image_text_from_dataset,
+    dtype.NDT,
+)
+op_class = fn.operator_add("ClipDataReader")
+op_class.__module__ = fn.__name__
+setattr(fn, "ClipDataReader", op_class)
```

Comment:
Is it planned to add an image-text reader to `habana_frameworks.mediapipe` in a future version of SynapseAI?

Reply:
No. Media reader is closely related to each task (image/text keys).  See [changes in clip reader](https://github.com/huggingface/optimum-habana/pull/256/files#diff-de4ae9309337ddf655cdd28451c1128420dca18cc3a7e4120222c98b201552aaR93).